worksheet_A/TerminalHacking.cpp

```diff
@@ -1,46 +1,152 @@
-	options.insert(secret);
-
-	// Fill the set with more words
-	// Using a set for options guarantees that the elements are all different
-	while (options.size() < numberOfWords)
-	{
-		std::string word = words.getRandomWord();
-		options.insert(word);
-	}
-
-	// Display all words
-	for each (std::string word in options)
-	{
-		std::cout << word << std::endl;
-	}
-
-	// TODO: implement the rest of the game
-
-    return 0;
-}
-
+// PartA_TerminalHacking.cpp : Defines the entry point for the console application.
+//
+
+#include "stdafx.h"
+#include "WordList.h"
+
+const int wordLength = 5;
+const int numberOfWords = 15;
+
+// Returns likness, the number of matching letters
+int getLikeness(std::string guessededWord, std::string secretWord)
+{
+	int result = 0;
+
+	for (int i = 0; i <= (secretWord.length() - 1); i ++)
+	{
+		if (secretWord.at(i) == guessededWord.at(i)) 
+		{
+			result += 1;
+		}
+	}
+
+	return result;
+}
+
+// Checks if word is in the list
+bool wordInList(std::string guessed, std::set<std::string> options)
+{
+	for each (std::string word in options) 
+	{
+		if (guessed == word) 
+		{
+			return 1;
+		}
+	}
+
+	return 0;
+}
+
+int main()
+{
+	// Seed the random number generator with the current time,
+	// to ensure different results each time the program is run
+	srand(static_cast<unsigned int>(time(nullptr)));
+
+	// Initialise word list
+	WordList words(wordLength);
+
+	// Choose secret word
+	std::string secret = words.getRandomWord();
+
+	// Create a set to hold the list of options
+	std::set<std::string> options;
+
+	// Put the secret word in the set
+	options.insert(secret);
+
+	// Fill the set with more words
+	// Using a set for options guarantees that the elements are all different
+	while (options.size() < numberOfWords)
+	{
+		std::string word = words.getRandomWord();
+		options.insert(word);
+	}
+
+	// Display all words
+	for each (std::string word in options)
+	{
+		std::cout << word << std::endl;
+	}
+
+	// TODO: implement the rest of the game
+	
+	using namespace std;
+
+	int attempts_remaining = 4;
+	string guessed;
+
+	cout << endl << "Welcome to ROBCO Industries (TM) Termlink" << endl << "Password Required";
+
+	// Main loop
+	while (attempts_remaining > 0) 
+			{
+				// Tells player how many attempts they have left
+				if (attempts_remaining > 1)
+				{
+					cout << endl << "Attempts Remaining: " << attempts_remaining << endl << endl;
+					cin >> guessed;
+					attempts_remaining -= 1;
+				}
+				else 
+				{
+					cout << endl << "Attempts Remaining: " << attempts_remaining << endl
+					<< "I Have a bad feeling about this" << endl << endl;
+					cin >> guessed;
+					attempts_remaining -= 1;
+				}
+
+				// Checks if word is in list
+				if (wordInList(guessed, options)) 
+				{									
+					// Checks if word is the secret word
+					if (guessed == secret) 
+					{
+						cout << "Password Accepted!" << endl << "The force is strong with you" << endl;
+						attempts_remaining = 0;

```

Comment:
Needs a comment saying that this causes the loop to end after this iteration. But maybe better to use the `break` keyword here.